cocotb/__init__.py

```diff
@@ -209,3 +210,162 @@ def process_plusargs():
+    """Set the level for the coroutine loggers.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.coroutine', lvl)
+
+def coroutineLogGetLevel():
+    """Returns the effective level coroutine loggers.
+    """
+    return _getLevel('cocotb.coroutine')
+
+def functionLogSetLevel(lvl):
+    """Set the level for the function loggers.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.function', lvl)
+
+def functionLogGetLevel():
+    """Returns the effective level function loggers.
+    """
+    return _getLevel('cocotb.function')
+
+def driverLogSetLevel(lvl):
+    """Set the level for the driver loggers.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.driver', lvl)
+
+def driverLogGetLevel():
+    """Returns the effective level driver loggers.
+    """
+    return _getLevel('cocotb.driver')
+
+def handleLogSetLevel(lvl):
+    """Set the level for the handle loggers.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.handle', lvl)
+
+def handleLogGetLevel():
+    """Returns the effective level handle loggers.
+    """
+    return _getLevel('cocotb.handle')
+
+def monitorLogSetLevel(lvl):
+    """Set the level for the monitor loggers.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.monitor', lvl)
+
+def monitorLogGetLevel():
+    """Returns the effective level monitor loggers.
+    """
+    return _getLevel('cocotb.monitor')
+
+def regressionLogSetLevel(lvl):
+    """Set the level for the regression logger.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.regression', lvl)
+
+def regressionLogGetLevel():
+    """Returns the effective level regression logger.
+    """
+    return _getLevel('cocotb.regression')
+
+def schedulerLogSetLevel(lvl):
+    """Set the level for the scheduler logger.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.scheduler', lvl)
+
+def schedulerLogGetLevel():
+    """Returns the effective level scheduler logger.
+    """
+    return _getLevel('cocotb.scheduler')
+
+def scoreboardLogSetLevel(lvl):
+    """Set the level for the scoreboard logger.
+
+    Args:
+        lvl (int):  Log level
+    """
+    _setLevel('cocotb.scoreboard', lvl)
+
+def scoreboardLogGetLevel():
+    """Returns the effective level scoreboard logger.
+    """
+    return _getLevel('cocotb.scoreboard')
+
+def _setLevel(name,lvl):
+    """Set the level for a logger.
+
+    Args:
+        name (str): Name of the logger
+         lvl (int): Log level
+    """
+    cfg = {
+        'version':1,
+        'incremental':True,
+        'loggers': {
+            name: {
+                'level': lvl
+            }
+        }
+    }
+    log.configure(cfg)
+
+def _getLevel(name=None):
+    """Returns the effective level of the a logger.
+
+    Args:
+        name (str): name of the logger
+    """
+    return log.SimLog(name=name).getEffectiveLevel()
```

Comment:
I don't really understand the purpose of these functions - can't we just expose the builtin logger objects directly to the user?